x-pack/plugins/security_solution/public/timelines/components/notes/add_note/index.tsx

```diff
@@ -47,46 +48,58 @@ export const AddNote = React.memo<{
   getNewNoteId: GetNewNoteId;
   newNote: string;
   onCancelAddNote?: () => void;
+  status: TimelineStatusLiteral;
   updateNewNote: UpdateInternalNewNote;
   updateNote: UpdateNote;
-}>(({ associateNote, getNewNoteId, newNote, onCancelAddNote, updateNewNote, updateNote }) => {
-  const handleClick = useCallback(
-    () =>
-      updateAndAssociateNode({
-        associateNote,
-        getNewNoteId,
-        newNote,
-        updateNewNote,
-        updateNote,
-      }),
-    [associateNote, getNewNoteId, newNote, updateNewNote, updateNote]
-  );
-
-  return (
-    <AddNotesContainer alignItems="flexEnd" direction="column" gutterSize="none">
-      <NewNote note={newNote} noteInputHeight={200} updateNewNote={updateNewNote} />
-      <EuiFlexItem grow={true}>
-        <MarkdownHint show={newNote.trim().length > 0} />
-      </EuiFlexItem>
-      <ButtonsContainer gutterSize="none">
-        {onCancelAddNote != null ? (
+}>(
+  ({
+    associateNote,
+    getNewNoteId,
+    newNote,
+    onCancelAddNote,
+    status,
+    updateNewNote,
+    updateNote,
+  }) => {
+    const handleClick = useCallback(
+      () =>
+        updateAndAssociateNode({
+          associateNote,
+          getNewNoteId,
+          newNote,
+          updateNewNote,
+          updateNote,
+        }),
+      [associateNote, getNewNoteId, newNote, updateNewNote, updateNote]
+    );
+    const hasNewNote = newNote.trim().length > 0;
+    const isImmutable = status === TimelineStatus.immutable;
+    return (
+      <AddNotesContainer alignItems="flexEnd" direction="column" gutterSize="none">
+        <NewNote note={newNote} noteInputHeight={200} updateNewNote={updateNewNote} />
+        <EuiFlexItem grow={true}>
+          <MarkdownHint show={hasNewNote} />
+        </EuiFlexItem>
+        <ButtonsContainer gutterSize="none">
+          {onCancelAddNote != null ? (
+            <EuiFlexItem grow={false}>
+              <CancelButton onCancelAddNote={onCancelAddNote} />
+            </EuiFlexItem>
+          ) : null}
           <EuiFlexItem grow={false}>
-            <CancelButton onCancelAddNote={onCancelAddNote} />
+            <EuiButton
+              data-test-subj="add-note"
+              isDisabled={!hasNewNote || isImmutable}
```

Comment:
I know we talked about doing that in another PR, but I think it will be pretty easy to do it here and hide the full form instead of just disabling the button. what do you think?

Reply:
ok, I've done that.